### PR TITLE
Allow empty responses instead of errors for filtered events

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: tests/30rooms/60anonymousaccess.pl [hunk 1]
```diff
@@ -756,3 +754,28 @@ sub ignore_presence_for
       )
    } @events ];
 }
+
+sub expect_4xx_or_empty_chunk
+{
+   my ( $f ) = @_;
```
Comment (leonerd) [inline tests/30rooms/60anonymousaccess.pl]: Wants linefeed after `@_`-unpacking line

File: tests/30rooms/60anonymousaccess.pl [hunk 2]
```diff
@@ -756,3 +754,28 @@ sub ignore_presence_for
...
+   $f->then( sub {
+      my ( $body ) = @_;
+
+      log_if_fail "Body", $body;
+
+      assert_json_keys( $body, qw( chunk ) );
+      assert_json_list( $body->{chunk} );
+      die "Want list to be empty" if @{ $body->{chunk} };
```
Comment (leonerd) [inline tests/30rooms/60anonymousaccess.pl]: I wonder if we want an `assert_json_empty_list` - I'll have a look and see how often this pattern comes up

File: tests/30rooms/60anonymousaccess.pl [hunk 3]
```diff
@@ -756,3 +754,28 @@ sub ignore_presence_for
...
+
+      Future->done( 1 );
+   },
+   http => sub {
+      my ( undef, undef, $response ) = @_;
+
+      log_if_fail "Response", $response;
```
Comment (leonerd) [inline tests/30rooms/60anonymousaccess.pl]: Put "HTTP response" in the message, as usually we call "Response" the decoded JSON body of a 200 OK message.

**Comment by leonerd** [comment]
LGTM besides two tiny fixes

File: tests/30rooms/60anonymousaccess.pl (hunk 3 above)
Comment (illicitonion) [inline tests/30rooms/60anonymousaccess.pl]: Done

**Pull request merged**